examples/single_mem_node/main.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate raft;
+
+use std::sync::mpsc::{self, RecvTimeoutError};
+use std::time::{Duration, Instant};
+use std::thread;
+use std::collections::HashMap;
+
+use raft::prelude::*;
+use raft::storage::MemStorage;
+
+type ProposeCallback = Box<Fn() + Send>;
+
+enum Msg {
+    Propose { id: u8, cb: ProposeCallback },
+    #[allow(dead_code)] Raft(Message),
+}
+
+// A simple example about how to use the raft library in Rust.
+fn main() {
+    // Create a storage for Raft, here we just use a simple memory storage.
+    // You need to build your own persistent storage in your production.
+    // Please check the Storage trait in src/storage.rs to see how to implement one.
+    let storage = MemStorage::new();
+
+    // Create the configuration for the Raft node.
+    let cfg = Config {
+        // The unique ID for the Raft node.
+        id: 1,
+        // The Raft node list.
+        // Mostly, the peers need to be saved in the storage
+        // and we can get them from Storage::initial_state function, so here
+        // you need to set empty.
+        peers: vec![1],
+        // Election tick is for how long the follower may campaign again after
+        // it doesn't receive any message from the leader.
+        election_tick: 10,
+        // Heartbeat tick is for how long the leader needs to send
+        // a heartbeat to keepalive.
+        heartbeat_tick: 3,
+        // The max size limits the max size of each append message, mostly, 1 MB is enough.
```

Comment:
-> The max size limits the max size of each appended message. Mostly, 1 MB is enough.